src/Common/src/System/Memory/PinnedSpanMemoryManager.cs

```diff
@@ -0,0 +1,36 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Buffers
+{
+    internal unsafe class PinnedSpanMemoryManager<T> : MemoryManager<T> where T : struct
```

Comment:
Nit: `Pointer` instead of `PinnedSpan`?  This type really doesn't care where the pointer and length came from.

Reply:
Fair point, changed (and changed the exception in Pin to NotSupportedException).